census/estimator/trainer/model.py

```diff
@@ -0,0 +1,246 @@
+      layers.embedding_column(occupation, dimension=embedding_size),
+      age,
+      education_num,
+      capital_gain,
+      capital_loss,
+      hours_per_week,
+  ]
+
+  return tf.contrib.learn.DNNLinearCombinedClassifier(
+      model_dir=model_dir,
+      linear_feature_columns=wide_columns,
+      dnn_feature_columns=deep_columns,
+      dnn_hidden_units=hidden_units or [100, 70, 50, 25])
+
+
+def parse_label_column(label_string_tensor): 
+  """Parses a string tensor into the label tensor
+  Args:
+    label_string_tensor: Tensor of dtype string. Result of parsing the
+    CSV column specified by LABEL_COLUMN
+  Returns:
+    A Tensor of the same shape as label_string_tensor, should return
+    an int64 Tensor representing the label index for classification tasks, 
+    and a float32 Tensor representing the value for a regression task.
+  """
+  # Build a Hash Table inside the graph
+  table = tf.contrib.lookup.string_to_index_table_from_tensor(
+      tf.constant(LABELS))
+
+  # Use the hash table to convert string labels to ints
+  return table.lookup(label_string_tensor)
+
+
+# ************************************************************************
+# YOU NEED NOT MODIFY ANYTHING BELOW HERE TO ADAPT THIS MODEL TO YOUR DATA
+# ************************************************************************
+
+
+def column_to_dtype(column):
+  if isinstance(column, layers.feature_column._SparseColumn):
+    return tf.string
+  else:
+    return tf.float32
+
+
+def serving_input_fn():
+    feature_placeholders = {
+        column.name: tf.placeholder(column_to_dtype(column), [None])
+        for column in INPUT_COLUMNS
+    }
+    features = {
+      key: tf.expand_dims(tensor, -1)
+      for key, tensor in feature_placeholders.items()
+    }
+    return input_fn_utils.InputFnOps(
+      features,
+      None,
+      feature_placeholders
+    )
+
+def generate_input_fn(filenames,
+                      num_epochs=None,
+                      shuffle=True,
+                      skip_header_lines=0,
+                      batch_size=40):
+  """Generates an input function for training or evaluation.
+  Args:
+      filenames: [str] list of CSV files to read data from.
+      num_epochs: int how many times through to read the data.
+        If None will loop through data indefinitely
+      shuffle: bool, whether or not to randomize the order of data.
+        Controls randomization of both file order and line order within
+        files.
+      skip_header_lines: int set to non-zero in order to skip header lines
+        in CSV files.
+      batch_size: int First dimension size of the Tensors returned by
+        input_fn
+  Returns:
+      A function () -> (features, indices) where features is a dictionary of
+        Tensors, and indices is a single Tensor of label indices.
+  """
+  def _input_fn():
+    filename_queue = tf.train.string_input_producer(
+        filenames, num_epochs=num_epochs, shuffle=shuffle)
```

Comment:
How do you want to choose between these two. If it's a one element list, match_filenames otherwise pass it explicitly? Seems like a reasonable approach.